components/canvas/utils/getProcessesNodes.tsx

```diff
@@ -0,0 +1,103 @@
+import type { NodeDataApi } from "@/types/NodeDataApi";
+import { NodeTypes } from "@/types/NodeTypes";
+import type { Project } from "@/types/Project";
+
+const createProjectNode = (project: Project): NodeDataApi => ({
+  id: project.vsmProjectID.toString(),
+  type: NodeTypes.project,
+  projectId: project.vsmProjectID.toString(),
+  index: null,
+  order: null,
+  description: "",
+  role: null,
+  duration: null,
+  unit: null,
+  children: [],
+  tasks: [],
+  linkedProject: null,
+  linkedProjectData: project,
+});
+
+const getLinkedNodes = (
+  apiNodes: NodeDataApi[],
+  node: NodeDataApi,
+  linkedProcesses: NodeDataApi[],
+  visited: Set<string> = new Set()
+): NodeDataApi[] => {
+  if (visited.has(node.id)) return linkedProcesses;
+  visited.add(node.id);
+
+  if (node.type === NodeTypes.linkedProcess) {
+    linkedProcesses.push({ ...node, children: [] });
+  }
+
+  node.children?.forEach((childId) => {
+    const childNode = apiNodes.find((n) => n.id === childId);
+    if (childNode) {
+      getLinkedNodes(apiNodes, childNode, linkedProcesses, visited);
+    }
+  });
+
+  return linkedProcesses;
+};
```

Comment:
I don't expect get functions to mutate arguments. This can be very confusing for developers that are not aware about it.

I understand this is a recursive function that needs mutability, but to make the usage a bit more safe, I would suggest hiding the recursive mutability in a subfunction.

```ts
const getLinkedNodes = (apiNodes: NodeDataApi[], node: NodeDataApi) => _getLinkedNodesRecursive(apiNodes, node, [])
```